addons/wiki/migrations/0009_migrate_node_wiki_pages.py

```diff
@@ -0,0 +1,195 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.9 on 2018-02-22 20:39
+from __future__ import unicode_literals
+
+import gc
+import logging
+import progressbar
+from django.db import migrations
+from django.db.models import Q
+from osf.utils.migrations import disable_auto_now_add_fields
+from django.contrib.contenttypes.models import ContentType
+from addons.wiki.models import WikiPage, NodeWikiPage, WikiVersion
+from osf.models import Comment, Guid, AbstractNode
+from bulk_update.helper import bulk_update
+logger = logging.getLogger(__name__)
+
+def reverse_func(state, schema):
+    """
+    Reverses NodeWikiPage migration. Repoints guids back to each NodeWikiPage,
+    repoints comment_targets, comments_viewed_timestamps, and deletes all WikiVersions and WikiPages
+    """
+    nwp_content_type_id = ContentType.objects.get_for_model(NodeWikiPage).id
+
+    nodes = AbstractNode.objects.exclude(wiki_pages_versions={})
+    progress_bar = progressbar.ProgressBar(maxval=nodes.count()).start()
+    for i, node in enumerate(nodes, 1):
+        progress_bar.update(i)
+        for wiki_key, version_list in node.wiki_pages_versions.iteritems():
+            if version_list:
+                for index, version in enumerate(version_list):
+                    nwp = NodeWikiPage.objects.filter(former_guid=version).include(None)[0]
+                    wp = WikiPage.load(version)
+                    guid = migrate_guid_referent(Guid.load(version), nwp, nwp_content_type_id)
+                    guid.save()
+                    nwp = guid.referent
+                move_comment_target(Guid.load(wp._id), wp, nwp)
+                update_comments_viewed_timestamp(node, wp, nwp)
+    progress_bar.finish()
+    WikiVersion.objects.all().delete()
+    WikiPage.objects.all().delete()
+    logger.info('NodeWikiPages restored and WikiVersions and WikiPages removed.')
+
+def move_comment_target(current_guid, current_target, desired_target):
+    """Move the comment's target from the current target to the desired target"""
+    desired_target_guid = Guid.load(desired_target._id)
+    if Comment.objects.filter(Q(root_target_id=current_guid) | Q(target=current_guid)).exists():
+        Comment.objects.filter(root_target=current_guid).update(root_target=desired_target_guid)
+        Comment.objects.filter(target=current_guid).update(target=desired_target_guid)
+    return
+
+def update_comments_viewed_timestamp(node, current_wiki_object, desired_wiki_object):
+    """Replace the current_wiki_object keys in the comments_viewed_timestamp dict with the desired wiki_object_id """
+    contributors_pending_save = []
+    for contrib in node.contributors.exclude(comments_viewed_timestamp={}):
+        if contrib.comments_viewed_timestamp.get(current_wiki_object._id, None):
```

Comment:
Since we're not auto-loading the GUID for `current_wiki_object`, this is going to result in a GUID query. Can you just pass in the GUID _id instead?